tests/phpunit/tests/rest-api/wpRestTemplateRevisionsController.php

```diff
@@ -0,0 +1,276 @@
+		);
+	}
+
+	/**
+	 * @covers WP_REST_Templates_Controller::get_items
+	 */
+	public function test_get_items() {
+		wp_set_current_user( self::$admin_id );
+		$request   = new WP_REST_Request(
+			'GET',
+			'/wp/v2/templates/' . self::TEST_THEME . '/' . self::TEMPLATE_NAME . '/revisions'
+		);
+		$response  = rest_get_server()->dispatch( $request );
+		$revisions = $response->get_data();
+
+		$this->assertCount(
+			4,
+			$revisions,
+			'Failed asserting that the response data contains exactly 4 items.'
+		);
+
+		$this->assertSame(
+			self::$template_post->ID,
+			$revisions[0]['parent'],
+			'Failed asserting that the parent ID of the revision matches the template post ID.'
+		);
+		$this->assertSame(
+			'Content revision #5',
+			$revisions[0]['content']['raw'],
+			'Failed asserting that the content of the revision is "Content revision #5".'
+		);
+
+		$this->assertSame(
+			self::$template_post->ID,
+			$revisions[1]['parent'],
+			'Failed asserting that the parent ID of the revision matches the template post ID.'
+		);
+		$this->assertSame(
+			'Content revision #4',
+			$revisions[1]['content']['raw'],
+			'Failed asserting that the content of the revision is "Content revision #4".'
+		);
+
+		$this->assertSame(
+			self::$template_post->ID,
+			$revisions[2]['parent'],
+			'Failed asserting that the parent ID of the revision matches the template post ID.'
+		);
+		$this->assertSame(
+			'Content revision #3',
+			$revisions[2]['content']['raw'],
+			'Failed asserting that the content of the revision is "Content revision #3".'
+		);
+
+		$this->assertSame(
+			self::$template_post->ID,
+			$revisions[3]['parent'],
+			'Failed asserting that the parent ID of the revision matches the template post ID.'
+		);
+		$this->assertSame(
+			'Content revision #2',
+			$revisions[3]['content']['raw'],
+			'Failed asserting that the content of the revision is "Content revision #2".'
+		);
+	}
+
+
+	public function test_get_item() {
+		wp_set_current_user( self::$admin_id );
+
+		// Choosing random revision for the test.
+		$revision_id = array_rand( wp_get_post_revisions( self::$template_post ) );
+
+		$request  = new WP_REST_Request( 'GET', '/wp/v2/templates/' . self::TEST_THEME . '/' . self::TEMPLATE_NAME . '/revisions/' . $revision_id );
+		$response = rest_get_server()->dispatch( $request );
+		$revision = $response->get_data();
+
+		$this->assertIsArray( $revision, 'Failed asserting that the revision is an array.' );
+		$this->assertSame(
+			$revision_id,
+			$revision['wp_id'],
+			sprintf(
+				'Failed asserting that the revision id is the same as %s.',
+				$revision_id
+			)
+		);
+		$this->assertSame(
+			self::$template_post->ID,
+			$revision['parent'],
+			sprintf(
+				'Failed asserting that the parent id of the revision is the same as %s.',
+				self::$template_post->ID
+			)
+		);
+	}
+
+	public function test_prepare_item() {
+	}
+
+	public function test_get_item_schema() {
+
+	}
```

Comment:
We need tests for both of these.

Reply:
Absolutely.

The version you reviewed was a work in progress. These tests have been added since your last review.